docs: document the 'Use *.dev.localhost URLs?' interactive prompt for `aspire new` (#697)
microsoft/aspire.dev: src/frontend/src/content/docs/reference/cli/commands/aspire-new.mdx

### Use \*.dev.localhost URLs?

During project creation, you may be prompted:

> **Use \*.dev.localhost URLs?**

This prompt determines whether the generated launch profile configures your Aspire services to be accessible via `*.dev.localhost` wildcard subdomain URLs (for example, `https://myservice.dev.localhost`) instead of standard `http://localhost` with port numbers.

#### What are \*.dev.localhost URLs?

`*.dev.localhost` is a wildcard subdomain pattern that resolves to your local machine (`127.0.0.1` or `::1`) in all modern browsers—no DNS or hosts-file configuration is required. The ASP.NET Core HTTPS development certificate includes `*.dev.localhost` as a Subject Alternative Name (SAN), so browsers trust connections to any subdomain under `*.dev.localhost` when the certificate is installed and trusted.

#### Yes — use \*.dev.localhost URLs

Choosing **Yes** configures the generated project so that each service gets a unique subdomain URL such as `https://myservice.dev.localhost`. This option:

- Enables HTTPS for all services from the start, leveraging the wildcard development certificate.

> - Gives each service its own origin, which prevents cookie and authentication token collisions between services running on the same machine.

Reviewer: @copilot please add the note about calling `aspire cert trust` and link to command ref with LearnMore

#### Yes — use \*.dev.localhost URLs

Choosing **Yes** configures the generated project so that each service gets a unique subdomain URL such as `https://myservice.dev.localhost`. This option:

- Enables HTTPS for all services from the start, leveraging the wildcard development certificate.
- Gives each service its own origin, which prevents cookie and authentication token collisions between services running on the same machine.
- More closely mirrors a production setup where services are hosted on separate domains or subdomains.

**Prerequisite:** The ASP.NET Core HTTPS development certificate must be installed and trusted on your machine. Run the following command if you haven't done so already:

```bash title="Aspire CLI"
aspire certs trust
```

<LearnMore>
  [Command reference: `aspire certs trust`](../aspire-certs-trust/)
</LearnMore>

<Aside type="note">

>   Safari on macOS does not resolve `*.dev.localhost` subdomains to

Reviewer: ```suggestion
```

  with Aspire templates as an alternative for project creation.
</Aside>

## Interactive prompts

When running `aspire new` in interactive mode, the command asks one or more questions before generating the project. The available prompts depend on the template and options you provide.

### Use \*.dev.localhost URLs?

During project creation, you may be prompted:

> **Use \*.dev.localhost URLs?**

This prompt determines whether the generated launch profile configures your Aspire services to be accessible via `*.dev.localhost` wildcard subdomain URLs (for example, `https://myservice.dev.localhost`) instead of standard `http://localhost` with port numbers.

> 

Reviewer 2: Nit: The Safari caveat is mentioned here *and* again in the `<Aside>` on line 73. Consider dropping the inline mention from this paragraph and relying solely on the callout — it's more prominent and avoids saying the same thing twice.

```suggestion
`*.dev.localhost` is a wildcard subdomain pattern that resolves to your local machine (`127.0.0.1` or `::1`) in most modern browsers without additional DNS configuration. The ASP.NET Core HTTPS development certificate includes `*.dev.localhost` as a Subject Alternative Name (SAN), so browsers trust connections to any subdomain under `*.dev.localhost` when the certificate is installed and trusted.
```

> 

Reviewer 2: Nit: This paragraph packs DNS resolution behavior, browser support exceptions, and HTTPS certificate details into one block. Consider splitting it into two paragraphs — one for DNS/browser resolution and one for the certificate trust story — to make it easier to scan.

> #### What are \*.dev.localhost URLs?

Reviewer 3: This is missing lots of nuance about what these are and aren't. The biggest being that it only works in browsers.

> `*.dev.localhost` is a wildcard subdomain pattern that resolves to your local machine (`127.0.0.1` or `::1`) in most modern browsers without additional DNS configuration. Safari on macOS is an exception and may require hosts-file entries for `*.dev.localhost` subdomains. The ASP.NET Core HTTPS development certificate includes `*.dev.localhost` as a Subject Alternative Name (SAN), so browsers trust connections to any subdomain under `*.dev.localhost` when the certificate is installed and trusted.

Reviewer 4: The **hosts** file shouldn't have a dash, but as it's a file name, it should be bold.:

```suggestion
`*.dev.localhost` is a wildcard subdomain pattern that resolves to your local machine (`127.0.0.1` or `::1`) in most modern browsers without additional DNS configuration. Safari on macOS is an exception and may require **hosts** file entries for `*.dev.localhost` subdomains. The ASP.NET Core HTTPS development certificate includes `*.dev.localhost` as a Subject Alternative Name (SAN), so browsers trust connections to any subdomain under `*.dev.localhost` when the certificate is installed and trusted.
```

#### Yes — use \*.dev.localhost URLs

Choosing **Yes** configures the generated project so that each service gets a unique subdomain URL such as `https://myservice.dev.localhost`. This option:

- Enables HTTPS for all services from the start, leveraging the wildcard development certificate.
- Gives each service its own origin, which prevents cookie and authentication token collisions between services running on the same machine.

> - More closely mirrors a production setup where services are hosted on separate (sub)domains.

Reviewer 4: ```suggestion
- More closely mirrors a production setup where services are hosted on separate domains or subdomains.
```

> **Prerequisite:** The ASP.NET Core HTTPS development certificate must be installed and trusted on your machine. Run the following command if you haven't done so already:

Reviewer 3: This is true, but I would just say developer cert and move all of the details about the cert to the cert article.

```bash title="Aspire CLI"
aspire certs trust
```

<LearnMore>
  [Command reference: `aspire certs trust`](../aspire-certs-trust/)
</LearnMore>

<Aside type="note">
  Safari on macOS does not resolve `*.dev.localhost` subdomains to
  `127.0.0.1` by default. If you use Safari, choose **No** or add an entry to

>   your hosts file.

Reviewer 4: ```suggestion
  your **hosts** file.
```

</Aside>

> #### No — use standard localhost URLs (default)

Reviewer 2: Nit: "add an entry to your hosts file" leaves the reader guessing what the entry should be. A brief inline example would make this immediately actionable:

```suggestion
<Aside type="note">
  Safari on macOS does not resolve `*.dev.localhost` subdomains to
  `127.0.0.1` by default. If you use Safari, choose **No** or add entries to
  your hosts file (for example, `127.0.0.1 myservice.dev.localhost`).
</Aside>
```
